AnkiDroid/src/main/java/com/ichi2/anki/provider/CardContentProvider.java

```
@@ -418,23 +428,19 @@ private JSONArray getDeckCountsFromDueTreeNode(Sched.DeckDueTreeNode deck){
     }
 
     @Override
-    public int update(Uri uri, ContentValues values, String selection,
-                      String[] selectionArgs) {
-        logProviderCall("update", uri);
+    public int update(Uri uri, ContentValues values, String selection, String[] selectionArgs) {
+        if (shouldBlockUpdate(uri)) {
+            throwSecurityException("update", uri);
+        }
         Collection col = CollectionHelper.getInstance().getCol(getContext());
         if (col == null) {
             return 0;
```

I think it would be good to call Timber here (or throw an exception?), otherwise the methods are failing silently.